### PR TITLE
feat(forest-cloud): add logs command to display logs

### PR DESCRIPTION
**Pull request by Scra3**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by Scra3** [comment]
Task linked: [CU-86bxjfqta Cloud - Story #1 - Provide log access to developer through cloud CLI (cloud-toolkit) `-n` command](https://app.clickup.com/t/86bxjfqta)

File: packages/forest-cloud/src/services/environment-variables.ts [hunk 1]
```diff
@@ -57,6 +57,19 @@ export function validateSubscriptionUrl(subscriptionUrl: string): void {
   validateUrl(subscriptionUrl, 'FOREST_SUBSCRIPTION_URL', ['wss:']);
 }
 
+export function validateMissingForestEnvSecret(
+  forestEnvSecret: string,
+  fromCommand: 'logs' | 'bootstrap',
+): void {
+  if (!forestEnvSecret) {
```
Comment (Thenkei) [inline packages/forest-cloud/src/services/environment-variables.ts:64]: 👌

File: packages/forest-cloud/src/commands/logs.ts [hunk 2]
```diff
@@ -61,11 +61,14 @@ const logMessage = (logger: Logger, { message }: { message: string }) => {
...
+      if (level === 'Info') return logger.info(base, timestamp);
 
-      return logger[levelToLog[level]](`${base}\n\t${log.error.message}\t${log.error.stack}`);
+      return logger[levelToLog[level]](
+        `${base}\n\t${log.error.message}\t${log.error.stack}`,
+        timestamp,
+      );
     }
 
     return logger[levelToLog[log.level]](log.message);
```
Comment (Thenkei) [inline packages/forest-cloud/src/commands/logs.ts:74]:
```
  return logger[levelToLog[log.level]](log.message, timestamp);
```

File: packages/forest-cloud/src/commands/logs.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        vars.FOREST_ENV_SECRET = options.envSecret || vars.FOREST_ENV_SECRET;
+        validateMissingForestEnvSecret(vars.FOREST_ENV_SECRET, 'logs');
+
+        validateEnvironmentVariables(vars);
+
+        const { logs } = await buildHttpServer(vars).getLogs(options.tail);
+
+        if (logs?.length > 0) {
+          logs.forEach(logMessage.bind(null, logger));
```
Comment (realSpok) [inline packages/forest-cloud/src/commands/logs.ts:112]: why do you need the bind here ? cant you just call logMessage(logger) ?

File: packages/forest-cloud/src/commands/logs.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+export default (program: Command, context: MakeCommands) => {
+  const { logger, getCurrentVersion, login, getEnvironmentVariables, buildHttpServer } = context;
+  program
+    .command('logs')
+    .option(
+      '-e, --env-secret <string>',
+      'Environment secret, you can find it in your environment settings. (env: FOREST_ENV_SECRET)',
+    )
+    .option('-n, --tail <integer>', 'Number of lines to show from the end of the logs')
```
Comment (realSpok) [inline packages/forest-cloud/src/commands/logs.ts:88]:
```suggestion
    .option('-n, --tail <integer>', 'Number of lines to show from the end of the logs. may return slightly less lines than expected in some cases')
```

File: packages/forest-cloud/src/commands/logs.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    .description('Display the logs of the published customizations')
```
Comment (realSpok) [inline packages/forest-cloud/src/commands/logs.ts:89]:
```suggestion
    .description('Display the logs of the customizations published on your agent')
```

File: packages/forest-cloud/src/commands/logs.ts [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  }
+};
+
+export default (program: Command, context: MakeCommands) => {
+  const { logger, getCurrentVersion, login, getEnvironmentVariables, buildHttpServer } = context;
+  program
+    .command('logs')
+    .option(
+      '-e, --env-secret <string>',
+      'Environment secret, you can find it in your environment settings. (env: FOREST_ENV_SECRET)',
```
Comment (realSpok) [inline packages/forest-cloud/src/commands/logs.ts:86]:
```suggestion
      'Environment secret, you can find it in your environment settings. (you can also pass it with environment variable FOREST_ENV_SECRET)',
```

File: packages/forest-cloud/src/commands/logs.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  validateEnvironmentVariables,
+  validateMissingForestEnvSecret,
+} from '../services/environment-variables';
+import HttpServer from '../services/http-server';
+import { loginIfMissingAuthAndReturnEnvironmentVariables } from '../shared';
+import { Logger, MakeCommands } from '../types';
+
+const levelToLog = {
+  Info: 'info',
+  Warn: 'warn',
```
Comment (Thenkei) [inline packages/forest-cloud/src/commands/logs.ts:16]: error 👀

File: packages/forest-cloud/src/commands/logs.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+          const baseMessage = `Requested ${tail} log${pluralize} in the last hour`;
+
+          if (logs.length === Number(tail)) {
+            logger.spinner.succeed(baseMessage);
+          } else {
+            logger.spinner.succeed(`${baseMessage}, but only ${logs.length} were found`);
+          }
+
+          logs.forEach(log => displayLog(logger, log));
+        } else logger.spinner.warn('No logs found in the last hour');
```
Comment (Thenkei) [inline packages/forest-cloud/src/commands/logs.ts:85]: 👌

**Review by Thenkei — APPROVED** [review]
- ✅&nbsp;Code review
  - ✅&nbsp; The PR tackle one subject only
- ✅&nbsp;Automatic tests
  - ✅&nbsp; Unit tests
- ✅&nbsp;Manual tests 🥳 
- ✅&nbsp;PR title
- ✅&nbsp;PR linked to the clickup task

**Pull request merged**